src/units/abilities.cpp

```diff
@@ -1706,24 +1706,41 @@ effect::effect(const unit_ability_list& list, int def, bool backstab, const_atta
 		if (!filter_base_matches(cfg, def))
 			continue;
 
-		if (const config::attribute_value *v = cfg.get("value")) {
-			int value = get_single_ability_value(*v, def, ability, list.loc(), att, [&](const wfl::formula& formula, wfl::map_formula_callable& callable) {
-				callable.add("base_value", wfl::variant(def));
-				return formula.evaluate(callable).as_int();
-			});
-
-			int value_cum = cfg["cumulative"].to_bool() ? std::max(def, value) : value;
-			assert((set_effect_min.type != NOT_USED) == (set_effect_max.type != NOT_USED));
-			if(set_effect_min.type == NOT_USED) {
-				set_effect_min.set(SET, value_cum, ability.ability_cfg, ability.teacher_loc);
-				set_effect_max.set(SET, value_cum, ability.ability_cfg, ability.teacher_loc);
-			}
-			else {
-				if(value_cum > set_effect_max.value) {
+		//is_cumulable is a boolean value who is true when abilities or weapon
+		//used is [leadership], [drains], [heal_on_hit], [heals] or [regenerate] type.
+		//while what is_cumulable or "cumulative" attribute are flase, "value"
+		//calculation type is usual, else , "value" is equivalent to "add" or "sub" atribute.
+		if(!is_cumulable || !cfg["cumulative"].to_bool()){
+			if (const config::attribute_value *v = cfg.get("value")) {
+				int value = get_single_ability_value(*v, def, ability, list.loc(), att, [&](const wfl::formula& formula, wfl::map_formula_callable& callable) {
+					callable.add("base_value", wfl::variant(def));
+					return formula.evaluate(callable).as_int();
+				});
+
+				int value_cum = cfg["cumulative"].to_bool() ? std::max(def, value) : value;
```

Comment:
This line tests the value of `cfg["cumulative"]`, but it's inside a conditional block that already checks `(!is_cumulable || !cfg["cumulative"].to_bool())`.

Under what circumstances could we have both `!is_cumulable` and yet also `cfg["cumulative"].to_bool() == true`? It shoulds like a tag that shouldn't support `cumulative=yes` having different behavior if given the value `cumulative=yes`.

Reply:
[chance_to_hit] is a case where is_cumulable is false but cumulative can be true, and cumulative =no when no writen in wml code, and i don't want change this.